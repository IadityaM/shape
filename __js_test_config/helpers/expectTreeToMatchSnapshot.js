import toJson from 'enzyme-to-json'

const expectTreeToMatchSnapshot = tree => {
  expect(toJson(tree)).toMatchSnapshot()
}

export default expectTreeToMatchSnapshot
