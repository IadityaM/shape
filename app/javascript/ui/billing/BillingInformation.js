import Section from '~shared/components/molecules/Section'

class BillingInformation extends React.Component {
  render() {
    return <Section title="Billing Information">hi</Section>
  }
}

export default BillingInformation
