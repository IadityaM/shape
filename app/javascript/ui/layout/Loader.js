import styled from 'styled-components'
import Spinner from 'react-spinkit'
import { Flex, Box } from 'reflexbox'

import v from '~/utils/variables'

const StyledSpinner = styled(Spinner)`
  &.sk-spinner {
    margin: 0 auto;
    vertical-align: middle;
    height: 100px;
    width: 100px;
  }
`

const Logo = () => (
  <Flex style={{ height: '50vh' }} align="center" justify="center">
    <Box>
      <StyledSpinner
        fadeIn="half"
        name="folding-cube"
        color={v.colors.teal}
      />
    </Box>
  </Flex>
)

export default Logo
