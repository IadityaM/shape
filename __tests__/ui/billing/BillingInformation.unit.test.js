import BillingInformation from '~/ui/billing/BillingInformation'
import { shallow } from 'enzyme'
import Loader from '~/ui/layout/Loader'
import expectTreeToMatchSnapshot from '#/helpers/expectTreeToMatchSnapshot'

describe('BillingInformation', () => {
  let apiStore, networkStore

  beforeEach(() => {
    apiStore = {
      currentUserOrganizationId: 123,
      currentUserOrganization: {
        active_users_count: 0,
        trial_users_count: 0,
        trial_ends_at: '01/01/2022',
        is_within_trial_period: true,
        price_per_user: 1.23,
        current_billing_period_start: '01/01/2015',
        current_billing_period_end: '01/31/2015',
      },
    }
    networkStore = {
      loadOrganization: jest.fn(() => Promise.resolve()),
      loadPlans: jest.fn(() => Promise.resolve()),
    }
  })
  describe('still within trial', () => {
    beforeEach(() => {
      apiStore.currentUserOrganization.is_within_trial_period = true
      apiStore.currentUserOrganization.trial_users_count = 10
      apiStore.currentUserOrganization.active_users_count = 5
    })
    it('renders trial information', async () => {
      const wrapper = shallow(
        <BillingInformation.wrappedComponent
          apiStore={apiStore}
          networkStore={networkStore}
        />
      )
      expect(wrapper.find(Loader).exists()).toBe(true)
      await Promise.all([networkStore.loadOrganization, networkStore.loadPlans])
      expect(networkStore.loadOrganization).toHaveBeenCalledWith(
        apiStore.currentUserOrganizationId
      )
      expect(networkStore.loadPlans).toHaveBeenCalledWith(
        apiStore.currentUserOrganizationId
      )
      wrapper.update()
      expectTreeToMatchSnapshot(wrapper)
    })
  })

  describe('trial expired', () => {
    beforeEach(() => {
      apiStore.currentUserOrganization.is_within_trial_period = false
      apiStore.currentUserOrganization.active_users_count = 20
    })

    it('renders without trial information', async () => {
      const wrapper = shallow(
        <BillingInformation.wrappedComponent
          apiStore={apiStore}
          networkStore={networkStore}
        />
      )
      expect(wrapper.find(Loader).exists()).toBe(true)
      await Promise.all([networkStore.loadOrganization, networkStore.loadPlans])
      expect(networkStore.loadOrganization).toHaveBeenCalledWith(
        apiStore.currentUserOrganizationId
      )
      expect(networkStore.loadPlans).toHaveBeenCalledWith(
        apiStore.currentUserOrganizationId
      )
      wrapper.update()
      expectTreeToMatchSnapshot(wrapper)
    })
  })

  describe('within trial, exceeded trial users', () => {
    beforeEach(() => {
      apiStore.currentUserOrganization.is_within_trial_period = true
      apiStore.currentUserOrganization.trial_users_count = 5
      apiStore.currentUserOrganization.active_users_count = 100
    })

    it('renders with trial and billing information', async () => {
      const wrapper = shallow(
        <BillingInformation.wrappedComponent
          apiStore={apiStore}
          networkStore={networkStore}
        />
      )
      expect(wrapper.find(Loader).exists()).toBe(true)
      await Promise.all([networkStore.loadOrganization, networkStore.loadPlans])
      expect(networkStore.loadOrganization).toHaveBeenCalledWith(
        apiStore.currentUserOrganizationId
      )
      expect(networkStore.loadPlans).toHaveBeenCalledWith(
        apiStore.currentUserOrganizationId
      )
      wrapper.update()
      expectTreeToMatchSnapshot(wrapper)
    })
  })
})
