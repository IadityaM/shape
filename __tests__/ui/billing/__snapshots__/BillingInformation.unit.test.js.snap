// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingInformation in app billing disabled renders with the in app billing notice 1`] = `
<styled.div>
  <Section
    title="Billing Information"
  >
    <WithStyles(Grid)
      container={true}
      justify="space-between"
    >
      <Block
        title="Current Active People"
      >
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="space-between"
        >
          <WithStyles(Grid)
            item={true}
          >
            <Typography
              variant="emphasis"
            >
              0
            </Typography>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            Total People
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <div>
          <styled.div>
            <Box
              my={20}
              p={10}
            >
              <WithStyles(Grid)
                container={true}
                justify="space-between"
              >
                <WithStyles(Grid)
                  item={true}
                >
                  0
                </WithStyles(Grid)>
                <WithStyles(Grid)
                  item={true}
                >
                  People on free trial
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </Box>
          </styled.div>
          <HorizontalDivider />
          <Box
            p={10}
          >
            <WithStyles(Grid)
              container={true}
              justify="space-between"
            >
              <WithStyles(Grid)
                item={true}
              >
                0
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                Billable people
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </Box>
        </div>
      </Block>
      <React.Fragment>
        <Block
          title="Current Monthly Rate"
        >
          <FancyDollarAmount>
            0
          </FancyDollarAmount>
          <Box
            my={25}
          >
            $1.23
            /person per month
          </Box>
          <styled.div>
            Total monthly rate does not include people covered by free trial conditions.
          </styled.div>
        </Block>
        <Block
          title="Current Billing Period"
        >
          <WithStyles(Grid)
            container={true}
            justify="space-between"
            spacing={16}
          >
            <WithStyles(Grid)
              item={true}
            >
              <styled.span>
                Start date:
              </styled.span>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              01/01/2015
            </WithStyles(Grid)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            container={true}
            justify="space-between"
            spacing={16}
          >
            <WithStyles(Grid)
              item={true}
            >
              <styled.span>
                End date:
              </styled.span>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              01/31/2015
            </WithStyles(Grid)>
          </WithStyles(Grid)>
          <div>
            <Styled(styled.div)>
              FREE TRIAL
            </Styled(styled.div)>
            <WithStyles(Grid)
              container={true}
              justify="space-between"
              spacing={16}
            >
              <WithStyles(Grid)
                item={true}
              >
                <styled.span>
                  Free trials used:
                </styled.span>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                0
                 of 
                0
              </WithStyles(Grid)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              container={true}
              justify="space-between"
              spacing={16}
            >
              <WithStyles(Grid)
                item={true}
              >
                <styled.span>
                  Trial expires:
                </styled.span>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                01/01/2022
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </div>
        </Block>
      </React.Fragment>
    </WithStyles(Grid)>
  </Section>
</styled.div>
`;

exports[`BillingInformation in app billing enabled still within trial renders trial information 1`] = `
<styled.div>
  <Section
    title="Billing Information"
  >
    <WithStyles(Grid)
      container={true}
      justify="space-between"
    >
      <Block
        title="Current Active People"
      >
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="space-between"
        >
          <WithStyles(Grid)
            item={true}
          >
            <Typography
              variant="emphasis"
            >
              5
            </Typography>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            Total People
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <div>
          <styled.div>
            <Box
              my={20}
              p={10}
            >
              <WithStyles(Grid)
                container={true}
                justify="space-between"
              >
                <WithStyles(Grid)
                  item={true}
                >
                  5
                </WithStyles(Grid)>
                <WithStyles(Grid)
                  item={true}
                >
                  People on free trial
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </Box>
          </styled.div>
          <HorizontalDivider />
          <Box
            p={10}
          >
            <WithStyles(Grid)
              container={true}
              justify="space-between"
            >
              <WithStyles(Grid)
                item={true}
              >
                0
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                Billable people
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </Box>
        </div>
      </Block>
      <React.Fragment>
        <Block
          title="Current Monthly Rate"
        >
          <FancyDollarAmount>
            0
          </FancyDollarAmount>
          <Box
            my={25}
          >
            $1.23
            /person per month
          </Box>
          <styled.div>
            Total monthly rate does not include people covered by free trial conditions.
          </styled.div>
        </Block>
        <Block
          title="Current Billing Period"
        >
          <WithStyles(Grid)
            container={true}
            justify="space-between"
            spacing={16}
          >
            <WithStyles(Grid)
              item={true}
            >
              <styled.span>
                Start date:
              </styled.span>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              01/01/2015
            </WithStyles(Grid)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            container={true}
            justify="space-between"
            spacing={16}
          >
            <WithStyles(Grid)
              item={true}
            >
              <styled.span>
                End date:
              </styled.span>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              01/31/2015
            </WithStyles(Grid)>
          </WithStyles(Grid)>
          <div>
            <Styled(styled.div)>
              FREE TRIAL
            </Styled(styled.div)>
            <WithStyles(Grid)
              container={true}
              justify="space-between"
              spacing={16}
            >
              <WithStyles(Grid)
                item={true}
              >
                <styled.span>
                  Free trials used:
                </styled.span>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                5
                 of 
                10
              </WithStyles(Grid)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              container={true}
              justify="space-between"
              spacing={16}
            >
              <WithStyles(Grid)
                item={true}
              >
                <styled.span>
                  Trial expires:
                </styled.span>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                01/01/2022
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </div>
        </Block>
      </React.Fragment>
    </WithStyles(Grid)>
  </Section>
</styled.div>
`;

exports[`BillingInformation in app billing enabled trial expired renders without trial information 1`] = `
<styled.div>
  <Section
    title="Billing Information"
  >
    <WithStyles(Grid)
      container={true}
      justify="space-between"
    >
      <Block
        title="Current Active People"
      >
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="space-between"
        >
          <WithStyles(Grid)
            item={true}
          >
            <Typography
              variant="emphasis"
            >
              20
            </Typography>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          />
        </WithStyles(Grid)>
      </Block>
      <React.Fragment>
        <Block
          title="Current Monthly Rate"
        >
          <FancyDollarAmount>
            24.6
          </FancyDollarAmount>
          <Box
            my={25}
          >
            $1.23
            /person per month
          </Box>
          <styled.div>
            Total monthly rate does not include people covered by free trial conditions.
          </styled.div>
        </Block>
        <Block
          title="Current Billing Period"
        >
          <WithStyles(Grid)
            container={true}
            justify="space-between"
            spacing={16}
          >
            <WithStyles(Grid)
              item={true}
            >
              <styled.span>
                Start date:
              </styled.span>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              01/01/2015
            </WithStyles(Grid)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            container={true}
            justify="space-between"
            spacing={16}
          >
            <WithStyles(Grid)
              item={true}
            >
              <styled.span>
                End date:
              </styled.span>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              01/31/2015
            </WithStyles(Grid)>
          </WithStyles(Grid)>
        </Block>
      </React.Fragment>
    </WithStyles(Grid)>
  </Section>
</styled.div>
`;

exports[`BillingInformation in app billing enabled within trial, exceeded trial users renders with trial and billing information 1`] = `
<styled.div>
  <Section
    title="Billing Information"
  >
    <WithStyles(Grid)
      container={true}
      justify="space-between"
    >
      <Block
        title="Current Active People"
      >
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="space-between"
        >
          <WithStyles(Grid)
            item={true}
          >
            <Typography
              variant="emphasis"
            >
              100
            </Typography>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
          >
            Total People
          </WithStyles(Grid)>
        </WithStyles(Grid)>
        <div>
          <styled.div>
            <Box
              my={20}
              p={10}
            >
              <WithStyles(Grid)
                container={true}
                justify="space-between"
              >
                <WithStyles(Grid)
                  item={true}
                >
                  5
                </WithStyles(Grid)>
                <WithStyles(Grid)
                  item={true}
                >
                  People on free trial
                </WithStyles(Grid)>
              </WithStyles(Grid)>
            </Box>
          </styled.div>
          <HorizontalDivider />
          <Box
            p={10}
          >
            <WithStyles(Grid)
              container={true}
              justify="space-between"
            >
              <WithStyles(Grid)
                item={true}
              >
                95
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                Billable people
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </Box>
        </div>
      </Block>
      <React.Fragment>
        <Block
          title="Current Monthly Rate"
        >
          <FancyDollarAmount>
            116.85
          </FancyDollarAmount>
          <Box
            my={25}
          >
            $1.23
            /person per month
          </Box>
          <styled.div>
            Total monthly rate does not include people covered by free trial conditions.
          </styled.div>
        </Block>
        <Block
          title="Current Billing Period"
        >
          <WithStyles(Grid)
            container={true}
            justify="space-between"
            spacing={16}
          >
            <WithStyles(Grid)
              item={true}
            >
              <styled.span>
                Start date:
              </styled.span>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              01/01/2015
            </WithStyles(Grid)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            container={true}
            justify="space-between"
            spacing={16}
          >
            <WithStyles(Grid)
              item={true}
            >
              <styled.span>
                End date:
              </styled.span>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              01/31/2015
            </WithStyles(Grid)>
          </WithStyles(Grid)>
          <div>
            <Styled(styled.div)>
              FREE TRIAL
            </Styled(styled.div)>
            <WithStyles(Grid)
              container={true}
              justify="space-between"
              spacing={16}
            >
              <WithStyles(Grid)
                item={true}
              >
                <styled.span>
                  Free trials used:
                </styled.span>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                5
                 of 
                5
              </WithStyles(Grid)>
            </WithStyles(Grid)>
            <WithStyles(Grid)
              container={true}
              justify="space-between"
              spacing={16}
            >
              <WithStyles(Grid)
                item={true}
              >
                <styled.span>
                  Trial expires:
                </styled.span>
              </WithStyles(Grid)>
              <WithStyles(Grid)
                item={true}
              >
                01/01/2022
              </WithStyles(Grid)>
            </WithStyles(Grid)>
          </div>
        </Block>
      </React.Fragment>
    </WithStyles(Grid)>
  </Section>
</styled.div>
`;
